Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -1798,14 +1800,347 @@ def remove_nulls_from_dictionary(dict):
+'Description': 'Malicious File'}}}
+
+    :type indicator: ``str``
+    :param indicator: Value (e.g. 8.8.8.8)
+
+    :type indicator_type: ``str``
+    :param indicator_type: e.g. 'IP'
+
+    :type vendor: ``str``
+    :param vendor: Integration ID
+
+    :type description: ``str``
+    :param description: Why it's malicious
+
+    :return: A malicious DBot entry
+    :rtype: ``dict``
+    """
+    file_types = ('md5', 'sha1', 'sha256', 'crc32', 'sha512', 'ctph')
+    indicator_type_lower = indicator_type.lower()
+    if indicator_type_lower in ('ip', 'email'):
+        key = 'Address'
+    elif indicator_type_lower == 'url':
+        key = 'Data'
+    elif indicator_type_lower == 'domain':
+        key = 'Name'
+    elif indicator_type_lower == 'cve':
+        key = 'ID'
+    elif indicator_type_lower in file_types:
+        key = indicator_type_lower.upper()
+        indicator_type_lower = 'file'
+    else:
+        raise DemistoException('Wrong indicator type supplied: {}'.format(indicator_type))
+    entry = {
+        key: indicator,
+        'Malicious': {
+            'Vendor': vendor,
+            'Description': description
+        }
+    }
+    return {outputPaths[indicator_type_lower]: entry}
+
+
+# Will add only if 'requests' module imported
+if 'requests' in sys.modules:
+    class BaseClient(object):
+        def __init__(self,
+                     server,
+                     base_suffix,
+                     integration_name,
+                     integration_command_name,
+                     integration_context_name,
+                     verify=True,
+                     proxy=False,
+                     ok_codes=None
+                     ):
+            """Base Client for use in integrations.
+
+             :type server: ``str``
+             :param server: Base server address
+
+             :type base_suffix: ``str``
+             :param base_suffix: suffix of API (e.g`/api/v2/`)
+
+             :type integration_name: ``str``
+             :param integration_name: Name as shown in UI (`Integration Name`)
+
+             :type integration_command_name: ``str``
+             :param integration_command_name: lower case with `-` divider (`integration-name`)
+
+             :type integration_context_name: ``str``
+             :param integration_context_name: camelcase with no dividers (`IntegrationName`)
+
+             :type verify: ``bool``
+             :param verify: Verify SSL
+
+             :type proxy: ``bool``
+             :param proxy: Use system proxy
+
+             :type ok_codes: ``tuple``
+             :param ok_codes: acceptable OK codes. in None will use response.ok
+
+             :return: No data returned
+             :rtype: ``None``
+             """
+            self._server = server.rstrip('/')
+            self._verify = verify
+            self._integration_name = str(integration_name)
+            self._integration_name_command = str(integration_command_name)
```

Comment:
read the comments from the other PR!!! then let me know when to review this section